### PR TITLE
New Cheat Sheet: John the Ripper

### PR DESCRIPTION
**Pull request by vduddu**
## Description of new Instant Answer, or changes

Commands and Cheat sheet for password cracking tool, John the Ripper

## Related Issues and Discussions

N/A

## People to notify

@vduddu

---

Instant Answer Page: https://duck.co/ia/view/{ID}

---
https://duck.co/ia/view/johntheripper_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@vduddu Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
@vduddu Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/johntheripper.json [hunk 1]
```diff
@@ -0,0 +1,152 @@
+{
+    "id": "johntheripper_cheat_sheet",
+    "name": "John the ripper",
+    "description": "Cheatsheet and command list of password cracking tool, John the ripper",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/johntheripper.json]: "Command list of password cracking tool, John the ripper" would be enough here

File: share/goodie/cheat_sheets/json/johntheripper.json [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    "metadata": {
+        "sourceName": "jtr-cheat-sheet",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/johntheripper.json]: Lets change sourcename to 
"Count Upon Security "

File: share/goodie/cheat_sheets/json/johntheripper.json [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+        "sourceUrl" : "https://countuponsecurity.files.wordpress.com/2016/09/jtr-cheat-sheet.pdf"
+    },
+
+    "aliases": [
+        "jtr", "johnny"
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/johntheripper.json]: is johnny a good alias? I think we can remove it.

File: share/goodie/cheat_sheets/json/johntheripper.json (hunk 2 above)
Comment (vduddu) [inline share/goodie/cheat_sheets/json/johntheripper.json]: Done :)

File: share/goodie/cheat_sheets/json/johntheripper.json [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                "val": "10 char"
+            },
+            {
+                "key": "--incremental:Alnum",
+                "val": "62 char"
+            }
+            
+        ],
+        
+        "Other Options": [
```
Comment (gautamkrishnar) [inline share/goodie/cheat_sheets/json/johntheripper.json]: Change this to `Other Commands`

**Comment by gautamkrishnar** [comment]
@vduddu Thanks for your contribution 🎉

**Pull request merged**